### PR TITLE
Unchecked iterator type

### PR DESCRIPTION
**Pull request by SleeplessByte**
You probably want to use the internal names() mapping provided by `JSONObject`. 
If you use a `StringBuilder`, probably want to append the strings to the buffer as opposed to creating 5 strings in the concatenation process and then adding that to the buffer.

I didn't actually test it.

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java [hunk 1]
```diff
@@ -270,32 +272,35 @@ public void sendLog(String log) {
...
+	private String convertJSONtoString(JSONObject json) throws JSONException {
 		StringBuilder result = new StringBuilder();
 		
 		if (json != null) {
-	        Iterator<String> iter = json.keys();
-	        if (iter.hasNext()) {
-	        	boolean first = true;
-	        	do {
+            JSONArray names = json.names();
+	        if ( names != null ) {
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java]: Can you stick with the Branch style convention we have and not insert spaces before and after the ( internals )?

Comment (SleeplessByte) [inline Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java]: Yeah, no problem!

Comment (SleeplessByte) [inline Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java]: 1424806

File: Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java [hunk 2]
```diff
@@ -270,23 +272,26 @@ public void sendLog(String log) {
 		}
 	}
 	
-	private String convertJSONtoString(JSONObject json) {
+	private String convertJSONtoString(JSONObject json) throws JSONException {
```
Comment (qinweigong) [inline Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java]: We are still catching JSONException down there, do we still need to throw it here?

Comment (SleeplessByte) [inline Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java]: Myself I really don't like to return null on an exception; surely you could return null instead.

File: Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java [hunk 3]
```diff
@@ -270,23 +272,26 @@ public void sendLog(String log) {
...
 		        		result.append("&");
 		        	}
-	        		
-	        		String key = iter.next();
-	        		String value;
+
+                    String value;
 	        		try {
 						value = json.getString(key);
 					} catch (JSONException e) {
```
Comment (qinweigong) [inline Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java]: With the method signature having "throws JSONException", we can remove the try/catch here?

**Pull request merged**